compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/FirArrayOfCallTransformer.kt

```diff
@@ -23,14 +24,15 @@ internal class FirArrayOfCallTransformer {
         get() {
             val function: FirCallableDeclaration<*> = getOriginalFunction() ?: return false
             return function is FirSimpleFunction &&
-                    // TODO: extend it to other variants, like emptyArray, intArrayOf, floatArrayOf, etc.
-                    function.name.asString() == "arrayOf" &&
+                    function.name.isArrayOf &&
```

Comment:
This check is not necessary since you're already checking the name in `checkParametersAndArguments`. (You also don't need `"emptyArray"` in `arrayOfNames`.)

Reply:
Aha, you're right. Thanks.